docs/api/tax-id-secrets.md

```diff
@@ -0,0 +1,329 @@
+
+// Delete secret
+err := vendorsv1alpha1.DeleteTaxIdSecret(ctx, client, vendor, taxIdRef)
+```
+
+## Best Practices
+
+### 1. **Secret Naming Convention**
+Use descriptive names that include the vendor name:
+```bash
+# Good
+acme-corp-tax-id
+john-doe-ssn
+international-corp-vat
+
+# Avoid
+tax-secret-1
+secret-abc
+```
+
+### 2. **Key Naming Convention**
+Use consistent key names within secrets:
+```yaml
+data:
+  tax-id: MTItMzQ1Njc4OQ==      # For EIN
+  ssn: MTIzLTQ1LTY3ODk=         # For SSN
+  vat-number: R0IxMjM0NTY3ODk=  # For VAT
+```
+
+### 3. **Namespace Strategy**
+- Use the same namespace as the vendor when possible
+- Specify namespace explicitly for cross-namespace references
+- Consider using a dedicated namespace for sensitive data
+
+### 4. **Labels and Annotations**
+Add proper labels for identification and management:
+```yaml
+metadata:
+  labels:
+    vendor.miloapis.com/vendor: acme-corp
+    vendor.miloapis.com/type: tax-id
+    vendor.miloapis.com/country: us
+  annotations:
+    vendor.miloapis.com/tax-id-type: ein
+    vendor.miloapis.com/created-by: admin
+```
+
+### 5. **Owner References**
+Set owner references to ensure cleanup:
+```yaml
+ownerReferences:
+  - apiVersion: vendors.miloapis.com/v1alpha1
+    kind: Vendor
+    name: acme-corp
+    uid: 12345678-1234-1234-1234-123456789abc
+    controller: true
+```
+
+## Migration from Plain Text
+
+If you have existing vendors with plain text tax IDs:
+
+1. **Create secrets** for each vendor's tax ID
+2. **Update vendor resources** to use `TaxIdRef` instead of `TaxId`
+3. **Verify secrets** are accessible and contain correct data
+4. **Test validation** to ensure everything works
+
+### Migration Script Example
+
+```bash
+#!/bin/bash
+# Extract tax IDs and create secrets
+kubectl get vendors -o json | jq -r '.items[] | select(.spec.taxInfo.taxId) | "\(.metadata.name) \(.spec.taxInfo.taxId)"' | while read vendor_name tax_id; do
+  # Create secret
+  kubectl create secret generic "${vendor_name}-tax-id" \
+    --from-literal=tax-id="$tax_id" \
+    --namespace=default
+  
+  # Add labels
+  kubectl label secret "${vendor_name}-tax-id" \
+    vendor.miloapis.com/vendor="$vendor_name" \
+    vendor.miloapis.com/type=tax-id
+done
+```
+
+## Security Considerations
+
+### 1. **RBAC Configuration**
+Ensure proper RBAC for secret access:
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: vendor-secret-reader
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "list"]
+  resourceNames: ["*-tax-id"]  # Restrict to tax ID secrets
+```
+
+### 2. **Network Policies**
+Consider network policies to restrict secret access:
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: restrict-secret-access
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          name: vendor-controllers
+```
```

Comment:
This shouldn't be needed? Standard k8s RBAC / Network Policies aren't used with Milo. The network policy APIs aren't even available.